File: LeetCode/3179-find-the-n-th-value-after-k-seconds/3179-find-the-n-th-value-after-k-seconds.py
class Solution:
    def valueAfterKSeconds(self, n: int, k: int) -> int:
        MOD = 1000000007
        ans = [1] * n
        for j in range(0, k):
            dp = [0] * n
            dp[0] = ans[0]
            for i in range(1, n):
                dp[i] = (dp[i-1] + ans[i]) % MOD
            ans = dp
        return ans[n-1]

File: LeetCode/3179-find-the-n-th-value-after-k-seconds/3179-find-the-n-th-value-after-k-seconds.java
class Solution {
    public int valueAfterKSeconds(int n, int k) {
        int[] ans = new int[n];
        int MOD = 1000000007;
        Arrays.fill(ans, 1);
        for (int j = 0; j < k; j++) {
            int[] dp = new int[n];
            dp[0] = ans[0];
            for (int i = 1; i < n; i++) {
                dp[i] = (dp[i-1] + ans[i]) % MOD;
            }
            ans = dp;
        }
        return ans[n-1];
    }
}